src/core.cpp

```diff
@@ -61,22 +118,60 @@ CVC::CVC(
         std::unique_ptr<DecisionEngine> decision_engine,
         std::vector<std::unique_ptr<Character>> characters,
         std::mt19937 random_generator) {
+    this->ticks_ = 0;
     this->random_generator_ = random_generator;
     this->decision_engine_ = std::move(decision_engine);
     this->characters_ = std::move(characters);
 }
 
+std::vector<Character *> CVC::GetCharacters() const {
+    std::vector<Character *> ret;//(this->characters_.size());
+    for(auto& character : this->characters_) {
```

Comment:
You're writing `auto& character` not `auto &character` but you've got `Character *` in the line above for the vector as oppose to `Character*`.  In general, i think putting the * and & with the types is becoming more common, so I'd bias towards keeping `auto&` and changing `Character *` to `Character*` everywhere.

Reply:
that seems legit thanks for the feedback